examples/release_highlights/plot_release_highlights_0_23_0.py

```diff
@@ -0,0 +1,160 @@
+                                       (cat_proc, ('feat0', 'feat2')))
+
+clf = make_pipeline(preprocessor, LogisticRegression())
+clf
+
+##############################################################################
+# Scalability and stability improvements to KMeans
+# ------------------------------------------------
+# The :class:`~sklearn.cluster.KMeans` estimator was entirely re-worked, and it
+# is now significantly faster and more stable. In addition, the Elkan algorithm
+# is now compatible with sparse matrices. The estimator uses OpenMP based
+# parallelism instead of relying on joblib, so the `n_jobs` parameter has no
+# effect anymore. For more details on how to control the number of threads,
+# please refer to our :ref:`parallelism` notes.
+import scipy
+from sklearn.cluster import KMeans
+from sklearn.datasets import make_blobs
+from sklearn.metrics import completeness_score
+
+X, y = make_blobs(random_state=rng)
+X = scipy.sparse.csr_matrix(X)
+X_train, X_test, _, y_test = train_test_split(X, y, random_state=rng)
+kmeans = KMeans(algorithm='elkan').fit(X_train)
+print(completeness_score(kmeans.predict(X_test), y_test))
+
+##############################################################################
+# Improvements to the histogram-based Gradient Boosting estimators
+# ----------------------------------------------------------------
+# Various improvements were made to
+# :class:`~sklearn.ensemble.HistGradientBoostingClassifier` and
+# :class:`~sklearn.ensemble.HistGradientBoostingRegressor`. These estimators
+# now support :ref:`sample weights <sw_hgbdt>`, and a new Poisson loss function
+# was implemented. Also, an automatic early-stopping criterion was added:
+# early-stopping is enabled by default when the number of samples exceeds 10k.
+# The following snippet illustrates the use of the Poisson loss.
+
+from sklearn.experimental import enable_hist_gradient_boosting  # noqa
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+X = rng.randn(n_samples, n_features)
+# positive integer target correlated with X[:, 5] with many zeros:
+y = rng.poisson(lam=np.exp(X[:, 5]) / 2)
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=rng)
+reg = HistGradientBoostingRegressor(loss='poisson', learning_rate=.01)
+reg.fit(X_train, y_train)
+print(reg.score(X_test, y_test))
+
+##############################################################################
+# Finally, users can now define :ref:`monotonic constraints
+# <monotonic_cst_gbdt>` to constrain the predictions based on the variations of
+# specific features. In the following example, we construct a target that is
+# generally positively correlated with the first feature, with some noise.
+# Applying monotoinc constraints allows the prediction to capture the global
+# effect of the first feature, instead of fitting the noise.
+from sklearn.inspection import plot_partial_dependence
+from matplotlib import pyplot as plt
+
+n_samples = 5000
```

Comment:
The number of samples is so large that the plotted samples look like a model. Using 500 is sufficient to better, to show individual samples in the plot.
```suggestion
n_samples = 500
```